src/test/regress/sql/pg17.sql

```diff
@@ -224,7 +221,56 @@ ORDER BY c.relname;
 
 DROP TABLE test_partitioned;
 ALTER EXTENSION citus DROP ACCESS METHOD fake_am;
-
 -- End of testing specifying access method on partitioned tables.
 
+-- MAINTAIN privilege tests
+
+CREATE ROLE regress_maintain;
+CREATE ROLE regress_no_maintain;
+
+ALTER ROLE regress_maintain WITH login;
+GRANT USAGE ON SCHEMA pg17 TO regress_maintain;
+ALTER ROLE regress_no_maintain WITH login;
+GRANT USAGE ON SCHEMA pg17 TO regress_no_maintain;
+
+SET citus.shard_count TO 1; -- For consistent remote command logging
+CREATE TABLE dist_test(a int, b int);
+SELECT create_distributed_table('dist_test', 'a');
+INSERT INTO dist_test SELECT i % 10, i FROM generate_series(1, 100) t(i);
+
+SET citus.log_remote_commands TO on;
+
+SET citus.grep_remote_commands = '%maintain%';
+GRANT MAINTAIN ON dist_test TO regress_maintain;
+RESET citus.grep_remote_commands;
+
+SET ROLE regress_no_maintain;
+-- Current role does not have MAINTAIN privileges on dist_test
+ANALYZE dist_test;
+VACUUM dist_test;
+
+SET ROLE regress_maintain;
+-- Current role has MAINTAIN privileges on dist_test
+ANALYZE dist_test;
+VACUUM dist_test;
+
+-- Take away regress_maintain's MAINTAIN privileges on dist_test
+RESET ROLE;
+REVOKE MAINTAIN ON dist_test FROM regress_maintain;
```

Comment:
Need a grep and reset grep here as well
```suggestion
SET citus.grep_remote_commands = '%maintain%';
REVOKE MAINTAIN ON dist_test FROM regress_maintain;
RESET citus.grep_remote_commands;
```